test/jetstream_test.go

```diff
@@ -8005,10 +8005,7 @@ func TestJetStreamAckExplicitMsgRemoval(t *testing.T) {
 				}
 				m.Respond(nil)
 
-				sseq, dseq, dc, _ := o2.ReplyInfo(m.Reply)
-				if dseq != uint64(i+3) {
-					t.Fatalf("Expected consumer sequence of %d got %d", i+3, dseq)
-				}
+				sseq, _, dc, _ := o2.ReplyInfo(m.Reply)
```

Comment:
I think checking only that message 3 and 4 (and no duplicate) are received. Again, checking `sseq == 3 && dc == 1` being a failure is wrong if it happens that 3 and 4 are delivered (and not redelivered) should timing be such that there is no redelivery.